File: FrontEnd/ticketbotproject/src/app/admin/components/model/approval.ts
import { ApprovalStatus } from "./approvalStatus";
import { Ticket } from "./ticket";
import { User } from "./user";

export interface Approval{
    approvalId:number;
    ticket:{
        ticketId:number;
        title:string;
        description:string;
        createdDate:string;
        status:string;
        priority:string;
        User:{
            userId:number;
            name:string;
            email:string;
            role:string
        };
    };
    approvedBy:{
        userId:number;
        name:string;
        email:string;
        role:string;
    };
    approvalDate:string;
    approvalStatus:ApprovalStatus;
}

File: Backend/approval-service/src/main/java/com/server/model/Approval.java
package com.server.model;

import java.time.LocalDateTime;

import com.server.enums.ApprovalStatus;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "approvals")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Approval {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long approvalId;

	@ManyToOne
	@JoinColumn(name = "ticket_id")
	private Ticket ticket;

	@ManyToOne
	@JoinColumn(name = "approved_by")
	private User approvedBy;

	@Column(name = "approval_date", nullable = false)
	private LocalDateTime approvalDate;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private ApprovalStatus approvalStatus;

}
